src/Illuminate/Database/Eloquent/Relations/MorphPivot.php

```diff
@@ -45,6 +45,10 @@ protected function setKeysForSaveQuery(Builder $query)
      */
     public function delete()
     {
+        if (isset($this->attributes[$this->getKeyName()])) {
+            return (int) parent::delete();
```

Comment:
Why is this cast needed?

Reply:
Hello, To be honest, I don't know.
I just took the code from the laravel 7.x branch since the use case worked there.

[MorphPivot Laravel 6.x](https://github.com/laravel/framework/blob/6.x/src/Illuminate/Database/Eloquent/Relations/MorphPivot.php#L46)

[MorphPivot Laravel 7.x](https://github.com/laravel/framework/blob/7.x/src/Illuminate/Database/Eloquent/Relations/MorphPivot.php#L46)